api/dataprocessor.go

```diff
@@ -113,6 +126,7 @@ func Fix(in []schema.Point, from, to, interval uint32) []schema.Point {
 			o -= 1
 		}
 	}
+	pointSlicePool.Put(in[:0])
```

Comment:
This seems backwards to me, I'd have thought that the caller would be responsible for returning `in` to the pool.  

Similarly, it seems like the caller should provide `out` so that it can be explicitly responsible for both getting it from the pool and cleaning it up afterwards.

Finally, is there a reason that we don't put the retry and allocation logic into `pointSlicePool`?

Reply:
> I'd have thought that the caller would be responsible for returning in to the pool.
Similarly, it seems like the caller should provide out so that it can be explicitly responsible for both getting it from the pool and cleaning it up afterwards.

we can do that. I definitely think the pool interaction for in and out should be consistent (either both in caller, or both within Fix).
so this would make Fix a more pure utility function, which is good. but the downside is pulling out the `neededCap := int((last-first)/interval + 1)` calculation out of Fix is a bit weird.

> Finally, is there a reason that we don't put the retry and allocation logic into pointSlicePool?

this makes sense. sidenote: I've been thinking of having separate size classes, that sort of stuff would all make sense if we make pointSlicePool a more full-featured "object" rather than merely a sync.Pool instance